apps/desktop/src/app/sync/sync.service.ts

```diff
@@ -392,4 +296,95 @@ export class SyncService {
       .pipe(switchMap((urls: string[]) => forkJoin(urls.map((url) => injectScript(url)))))
       .subscribe();
   }
+
+  getKb(slug: string): Observable<WritableKnowledgeBox> {
+    return this.sdk.currentAccount.pipe(
+      take(1),
+      switchMap((account) => this.sdk.nuclia.db.getKnowledgeBox(account.slug, slug)),
+    );
+  }
+
+  getNucliaKey(kb: WritableKnowledgeBox): Observable<{ token: string; kbid: string }> {
+    return kb
+      .createKeyForService(
+        {
+          title: 'Desktop',
+          role: 'SCONTRIBUTOR',
+        },
+        this.getExpirationDate(),
+      )
+      .pipe(map((token) => ({ ...token, kbid: kb.id })));
+  }
+
+  private getExpirationDate(): string {
+    const date = new Date();
+    date.setFullYear(date.getFullYear() + 1);
+    return Math.floor(date.getTime() / 1000).toString();
+  }
+
+  serverStatus(server: string): Observable<{ running: boolean }> {
+    return this.http.get<{ running: boolean }>(`${server}/status`).pipe(catchError(() => of({ running: false })));
+  }
+
+  setSyncServer(server: { url?: string; local?: boolean }) {
+    if (server.local) {
+      this._syncServer.next(LOCAL_SYNC_SERVER);
+      localStorage.setItem(SYNC_SERVER_KEY, LOCAL_SYNC_SERVER);
+      const electron = (window as any)['electron'];
+      if (electron) {
+        this.serverStatus(LOCAL_SYNC_SERVER)
+          .pipe(take(1))
+          .subscribe((res) => {
+            if (!res.running) {
+              electron.startLocalServer();
+            }
+          });
+      }
+    } else if (server.url) {
+      localStorage.setItem(SYNC_SERVER_KEY, server.url);
+      this._syncServer.next(server.url);
+    }
+  }
+
+  resetSyncServer() {
+    this._syncServer.next('');
+  }
+
+  authenticateToSource(source: ISourceConnector): Observable<boolean> {
+    return source.authenticate().pipe(
+      filter((authenticated) => authenticated),
+      take(1),
+      switchMap(() => {
+        return this.currentSource.pipe(
+          take(1),
+          switchMap((currentSource) =>
+            this.setSourceData(this.getCurrentSourceId(), { ...currentSource, data: source.getParametersValues() }),
+          ),
+          map(() => true),
+        );
+      }),
+    );
+  }
+
+  getCurrentSourceId(): string {
+    return localStorage.getItem(SOURCE_NAME_KEY) || '';
+  }
+
+  setCurrentSourceId(id: string) {
+    if (id) {
+      localStorage.setItem(SOURCE_NAME_KEY, id);
+    } else {
+      localStorage.removeItem(SOURCE_NAME_KEY);
+    }
+  }
```

Comment:
do you expect an empty string? the signature tends to feel like you will never get inside the else block.